lib/docker-sync/sync_manager.rb

```diff
@@ -121,10 +123,16 @@ def run(sync_name = nil)
 
     def join_threads
       begin
+        has_threads = false
         @sync_processes.each do |sync_process|
-          sync_process.watch_thread.join unless sync_process.watch_thread.nil?
+          if sync_process.watch_thread
+            sync_process.watch_thread.join
+            has_threads = true
+          end
         end
 
+        sleep unless has_threads
```

Comment:
well we have to start thinking if we start working with pid / pid files whatever solutions are common in ruby.